stock_valued_picking_report/models/stock_pack_operation.py

```diff
@@ -53,42 +54,43 @@ def _compute_sale_order_line_fields(self):
 
     @api.multi
     def sale_lines_values(self, sale_lines):
-        if len(sale_lines) <= 1:
-            price_unit = sale_lines.price_unit
-            discount = sale_lines.discount
-        else:
-            sum_qty = 0.0
-            sum_price = 0.0
-            sum_discount = 0.0
-            for sale_line in sale_lines:
-                sum_qty += sale_line.product_uom_qty
-                sum_price += sale_line.price_unit * sale_line.product_uom_qty
-                sum_discount += sale_line.discount * sale_line.product_uom_qty
-            price_unit = sum_price / (sum_qty or 1)
-            discount = sum_discount / (sum_qty or 1)
-        price_reduce = price_unit * (1 - (discount or 0.0) / 100.0)
+        sum_qty = 0.0
+        sum_price = 0.0
+        sum_discount = 0.0
+        sum_amount_untaxed = 0.0
+        sum_amount_tax = 0.0
+        sum_amount_total = 0.0
+        for sale_line in sale_lines:
+            sum_qty += sale_line.product_uom_qty
+            sum_price += sale_line.price_unit * sale_line.product_uom_qty
+            sum_discount += sale_line.discount * sale_line.product_uom_qty
+            amount_untaxed = sale_line.price_subtotal
+            sum_amount_untaxed += amount_untaxed
+            rm = sale_line.order_id.company_id.tax_calculation_rounding_method
+            if rm == 'round_globally':
+                price = sale_line.price_unit * (1 - (sale_line.discount or
+                                                     0.0) / 100)
+                taxes = sale_line.tax_id.compute_all(
+                    price,
+                    sale_line.order_id.currency_id,
+                    sale_line.product_uom_qty,
```

Comment:
What happens if the order is not served completely?

Reply:
In that case the value is not computed partially, I can try to use qty_delivered instead.